pkg/database/import.go

```diff
@@ -0,0 +1,166 @@
+package database
+
+import (
+	"github.com/rotisserie/eris"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+// Import will copy the contents of input db to output db.
+func Import(input, output *DbInstance, dryRun bool) error {
+	in := input.DB
+	if err := output.DB.Transaction(func(out *gorm.DB) error {
+		if err := importTable(in, out, dryRun, Experiment{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, ExperimentTag{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Run{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Tag{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Metric{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, LatestMetric{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Param{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Dashboard{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, App{}); err != nil {
+			return err
+		}
+
+		return nil
+	}); err != nil {
+		return eris.Wrapf(err, "error importing table data")
+	}
+
+	return nil
+}
+
+// importTable will copy the contents of one table (model) from sourceDB to destDB.
+func importTable[T any](sourceDB, destDB *gorm.DB, dryRun bool, model T) error {
+	// Query data from the source database
+	log.Infof("Importing %T", model)
+	rows, err := sourceDB.Model(&model).Rows()
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	log.Infof("Transferring records (dry run? %v)", dryRun)
+
+	// Stream data to the destination database in bulk with collision handling
+	for rows.Next() {
+		var item T
+		sourceDB.ScanRows(rows, &item)
+		found, err := findCollision(destDB, item)
```

Comment:
just an idea: what about using `ON CONFLICT` and `RETURNING` to deal with collisions directly at the DB level?

Reply:
I was assuming that we might have cases of conflict that the constraints wouldn't catch, eg experiment.ID is different but experiment.Name is conflicting. But I think you're right, i'll adapt for that.

Reply:
Also, from what I am reading, we have to do ON CONFLICT UPDATE for returning to work -- so some columns might be clobbered, eg Tag value

Reply:
ah crap, we do not want `ON CONFLICT UPDATE`!

Reply:
btw, for experiments, as the IDs are sequential, we want to rewire that (e.g. we create an experiment with the given name and a new ID, and then we insert the corresponding runs with the new ID)

Reply:
which is return the rows that could be inserted, which means you can then compare the length, and if needed diff it with the input to know which ones were duplicates

Reply:
re-numbering the IDs with FK refs seems non trivial

Reply:
now merging same-name experiments and renumbering experiment_id PK and FK